pkg/scheduler/ugm/group_tracker_test.go

```diff
@@ -158,44 +171,98 @@ func TestGTDecreaseTrackedResource(t *testing.T) {
 		t.Errorf("new resource create returned error or wrong resource: error %t, res %v", err, usage3)
 	}
 
-	removeQT, decreased = groupTracker.decreaseTrackedResource(hierarchy2, TestApp2, usage5, true)
+	removeQT, decreased = groupTracker.decreaseTrackedResource(path2, TestApp2, usage5, true)
 	if !decreased {
-		t.Fatalf("unable to decrease tracked resource: queuepath %+q, app %s, res %v, error %t", hierarchy2, TestApp2, usage2, err)
+		t.Fatalf("unable to decrease tracked resource: queuepath %s, app %s, res %v, error %t", path2, TestApp2, usage2, err)
 	}
 	assert.Equal(t, 0, len(groupTracker.getTrackedApplications()))
 	assert.Equal(t, removeQT, true, "wrong remove queue tracker value")
 }
 
-func TestGTSetMaxLimits(t *testing.T) {
+func TestGTSetAndClearMaxLimits(t *testing.T) {
```

Comment:
Note: original unit test did not properly verify the behavior of the tracker, it only checked the return value of the increment which is always true (this should be changed in a follow-up JIRA). Now we canRunApp & headroom is validated with different settings.